### ISSUE TITLE
Students review 

### ISSUE THREAD

**Issue by Bejagui** [jsxgraph/moodle-filter_jsxgraph#19]
Like GeoGebra plugin,  when students review their answers, random variables restart; so they can't check why they are wrong.

**Comment by andreas-web** [comment]
Good morning Bejagui!
I don't think I fully understand the problem yet. Would you like a construction to display the changes that were made the first time it was called up in the browser when it was called up for the second time? So the construction should save its state?
This is not due to the filter. Even with the manual integration of the jsxgraphcore.js, the JavaScript is reloaded when the page is called up again (which is the whole point of JavaScript). It shows again the state of the construction that the programmer (teacher) has determined.
It is possible to save part of the status in the database: With the plugin formulas the current status of the construction is saved when using global variables. Read also: https://github.com/jsxgraph/moodleformulas_jsxgraph
The plug-in for this is already included in the latest filter version.
Best wishes,
Andreas

**Comment by Bejagui** [comment]
Hi Andreas.
The example y attach, is downloaded from Formula qtype course. (I put the code and I attach the results)

![Answer](https://user-images.githubusercontent.com/86538808/124688149-e8aed080-de9b-11eb-847f-74f24d32ff94.png)
![Accept buttom](https://user-images.githubusercontent.com/86538808/124688146-e8aed080-de9b-11eb-8ea7-eaf437568653.png)
![Review](https://user-images.githubusercontent.com/86538808/124688150-e9476700-de9b-11eb-9814-fc791bdee020.png)

code:  

<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>

<h3 style="margin-top:5px;">Parábola</h3>
<p>Usa los deslizadores para indicar la curva de la ecuación: {a}x<sup>2</sup> + {b}x + {c} = 0:</p>

<div id="AT" class="jxgbox" style="width:500px; height:500px; background-color:#FFFDE1">


<jsxgraph width="500" height="500" box="AT">
(function(myFunction){
    var ax = sessionStorage.getItem("as");
    var bx = sessionStorage.getItem("bs");
    var cx = sessionStorage.getItem("cs");

    var Plano = JXG.JSXGraph.initBoard('AT', 
               {
                  boundingbox: [-10.5, 10.5, 10.5, -10.5], 
                  axis:true, 
                  showCopyright: false, 
                  shownavigation: false,   
                  zoom: {enabled: false, wheel: false },  
                  pan: { enabled: false, needTwoFingers: false }
               } ),

    a = Plano.create('slider',[[2,-5],[7,-5],[-5,ax,5]], {snapWidth:0.25, name:'a'}),
    b = Plano.create('slider',[[2,-6],[7,-6],[-5,bx,5]], {snapWidth:0.5, name:'b'}),
    c = Plano.create('slider',[[2,-7],[7,-7],[-5,cx,5]], {snapWidth:0.5, name:'c'}),
    f = Plano.create('functiongraph',[function(x){
            return a.Value()*x*x + b.Value()*x + c.Value();
        }]),

g = setInterval(function(){
      var aa = a.Value();
      var bb = b.Value();
      var cc = c.Value();
      $("input.formulas_number:eq(0):text").val(aa);
      $("input.formulas_number:eq(1):text").val(bb);
      $("input.formulas_number:eq(2):text").val(cc);
  }, 1000);
   })();

</jsxgraph>

 </div>
<p></p>


<script>
$(window).on( "load", function() {

        $(".formulas_number").css("width", "40px");           
        $(".formulas_number").css("height", "24px");
        $(".formulas_number").css("padding", "3");
        $(".formulas_number").css("background-color", "#FFF5D5");
        $(".formulas_number").css("border", "1px solid #1A3BA2");
        $(".formulas_number").css("border-radius", "5px");
        $(".formulas_number").css("outline", "none");
        $(".formulas_number").css("text-align", "center");
        $(".formulas_number").css("font-weight", "700");
        $(".formulas_number").css("font-size", "16px");


    $("input.submit").on("focus", function(){
        az=$("input.formulas_number:eq(0):text").val();
        bz=$("input.formulas_number:eq(1):text").val();
        cz=$("input.formulas_number:eq(2):text").val();
        sessionStorage.setItem("as", az);
        sessionStorage.setItem("bs", bz);
        sessionStorage.setItem("cs", cz);
    });
});
</script>

<button style="margin:10px 0 0 65%;" onclick="sessionStorage.clear();">Reiniciar</button>
<p>Usa los deslizadores para seleccionar los coeficientes.</p>
{#1}{#2}{#3}

**Comment by Bejagui** [comment]
In addition, Andreas, can you give me a link where I can to learn about pass of variables  JSXGraph-Formula-JSXGraph

Thanks for your time and attention.

**Comment by andreas-web** [comment]
Dear Bejagui!

As already mentioned, it is quite normal that the page is reloaded when "submitting the form". This will also restart and execute JavaScript.
How to use variables in fomulas is explained in this example here:
https://github.com/jsxgraph/moodleformulas_jsxgraph/blob/master/README.md#example

**Comment by andreas-web** [comment]
Hello again!
We came up with another possibility: It might be an alternative to save the data in localStorage on the user's device (like a cookie) and to read it out again after reloading. Read about the commands here: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage
Best regards,
Andreas

**Comment by Bejagui** [comment]
Very good; thank you
